### PR TITLE
12.x update to laravel11

### PR DESCRIPTION
**Pull request by kienngd**
## Description
**Pleas upgrade apiato/core project too**
composer.json

Add repository to my new github repository:
Add apiato/core: ^9.0.0. **Please change path of apiato_core package**
laravel/framework from ^10.0 to ^11.0
Add laravel/passport: ^12.0
nunomaduro/collision from ^7.0 to ^8.0
Add barryvdh/laravel-ide-helper: ^3.0
Add barryvdh/laravel-debugbar: ^3.13.5
composer.lock

Delete and re-install make it change.
app/Containers/AppSection/Authentication/composer.json

laravel/passport from ^11.0.0 to ^12.0.0
app/Containers/AppSection/Authentication/Configs/appSection-authentication.php

Add intval to email_verification_link_expiration_time_in_minute to work with new carbon
app/Containers/AppSection/Authorization/composer.json

spatie/laravel-permission from ^5.0 to ^6.0
app/Containers/AppSection/Authorization/Data/Migrations/2016_12_29_201047_create_permission_tables.php

Fix PermissionRegistrar::$pivotPermission, PermissionRegistrar::$pivotRole in new version of passport is not static.
app/Containers/AppSection/Authorization/Tests/Unit/Data/Migrations/MigrationTest.php

Change colume type to fix with mysql 8.2
app/Containers/AppSection/User/Tests/Unit/Data/Migrations/MigrationTest.php

Change colume type to fix with mysql 8.2
app/Ship/composer.json

Move all require and require-dev to root composer.json file

### CODE REVIEW

**Comment by innoflash** [comment]
@kienngd dont you think the core needs to update to Laravel 11 first before the app?

File: composer.json [hunk 1]
```diff
@@ -33,17 +33,25 @@
             "role": "Maintainer"
         }
     ],
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/kienngd/apiato_core.git"
```
Comment (polarikus) [inline composer.json:39]: WARN

Fork, not a link to the official repository

Comment (kienngd) [reply composer.json:39]: Because of apiato core need to upgrade laravel 11 too. I fork it, upgrade and make pull request for apiato core.

**Comment by Mohammad-Alavi** [comment]
Thank you for the effort you put into this @kienngd. 🌹 
Unfortunatelly a Laravel upgrade is not as easy as just updating the dependencies. We have to consider many other things like breaking changes, config changes, etc...

But your upgrade steps definitelly helps. I'll use it when I am going to upgrate Apiato to Laravel v11. 🥇

**Comment by kienngd** [comment]
Thank you for your reply.

**Pull request closed without merging**